modules/diskq/logqueue-disk-non-reliable.c

```diff
@@ -208,15 +180,23 @@ _move_messages_from_overflow(LogQueueDiskNonReliable *self)
 static void
 _move_messages_from_disk_to_qout(LogQueueDiskNonReliable *self)
 {
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+
   do
     {
-      LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
-      LogMessage *msg = _get_next_message(self, &path_options);
+      if (qdisk_get_length(self->super.qdisk) <= 0)
+        break;
+
+      LogMessage *msg = log_queue_disk_read_message(&self->super, &path_options);
 
       if (!msg)
         break;
 
-      _add_message_to_qout(self, msg, &path_options);
+      /* NOTE: we always generate flow-control disabled entries into
+       * qout, they only get there via backlog rewind */
+      g_queue_push_tail(self->qout, msg);
+      g_queue_push_tail(self->qout, LOG_PATH_OPTIONS_FOR_BACKLOG);
```

Comment:
This `LogPathOptions` should be acquired from `log_queue_disk_read_message()`. Unfortunately it does not fill it, currently, but I have made a commit which fixes that: https://github.com/syslog-ng/syslog-ng/pull/3754/commits/b2f63aca25754f5bf6cda9af6ff8e8e64823cea0

If that commit gets merged sooner, you can simply use the `path_options` variable here, or if this get merges sooner, it would be nice to modify this bit accordingly later.

Related discussion under another PR: https://github.com/syslog-ng/syslog-ng/pull/3754#discussion_r688510178